source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,446 @@
+$script:dscResourceCommonPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\DscResource.Common'
+$script:configMgrResourcehelper = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\ConfigMgrCBDsc.ResourceHelper'
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+ <#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy.
+
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName
```

Comment:
don't know if SiteServerName should be a required param as it isn't used for any of the command below

Reply:
if I make it not required, I may need to return false if specified and mismatch. commenting here to keep track

Reply:
Plan to make it not required, return true in test if mismatch, but output a message saying that the role is installed, just not on the specified server.

Reply:
changed and pushed